rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/object/MetaFieldGenerator.xtend

```diff
@@ -84,37 +84,43 @@ class MetaFieldGenerator {
 		
 		//find all the reference types
 		val namespaceClasses = Multimaps.index(modelClasses, [c|c.namespace]).asMap
-		for (nsc: namespaceClasses.entrySet) {
+		for (nsc : namespaceClasses.entrySet) {
 			if (ctx.cancelIndicator.canceled) {
 				return
 			}
-			val refs = nsc.value.filter(Data).flatMap[expandedAttributes].filter[hasMetas && metas.exists[name=="reference" || name=="address"]].toSet
-			
-			for (ref:refs) {
-				val targetModel = ref.type.model
-				if (targetModel.shouldGenerate) {
+			val refs = nsc.value.filter(Data).flatMap[expandedAttributes].filter [
+				hasMetas && metas.exists[name == "reference" || name == "address"]
+			].toSet
+
+			if (nsc.value.head.model.shouldGenerate) {
```

Comment:
I think we can get rid of these checks altogether. This check is already performed by the caller (`RosettaGenerator`)

Reply:
done